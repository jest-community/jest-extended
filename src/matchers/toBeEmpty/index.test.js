import matcher from './';

expect.extend(matcher);

describe('.toBeEmpty', () => {
  it('passes when given empty string', () => {
    expect('').toBeEmpty();
  });

  it('passes when given empty string object', () => {
    expect(new String('')).toBeEmpty();
  });

  it('passes when given empty array', () => {
    expect([]).toBeEmpty();
  });

  it('passes when given empty object', () => {
    expect({}).toBeEmpty();
  });

  it('fails when given non-empty string', () => {
    expect(() => expect('string').toBeEmpty()).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toBeEmpty', () => {
  it('passes when given a non-empty string', () => {
    expect('string').not.toBeEmpty();
  });

  it('passes when given a non-empty string object', () => {
    expect(new String('string')).not.toBeEmpty();
  });

  it('passes when given a non-empty array', () => {
    expect([1, 2]).not.toBeEmpty();
  });

  it('passes when given a non-empty object', () => {
    expect({ foo: 'bar' }).not.toBeEmpty();
  });

  it('fails when given empty string', () => {
    expect(() => expect('').not.toBeEmpty()).toThrowErrorMatchingSnapshot();
  });
});
