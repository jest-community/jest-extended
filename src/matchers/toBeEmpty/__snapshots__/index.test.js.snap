// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toBeEmpty fails when given empty string 1`] = `
"<dim>expect(</><red>received</><dim>).not.toBeEmpty(</><dim>)</>

Expected value to not be empty received:
  <red>\\"\\"</>"
`;

exports[`.toBeEmpty fails when given non-empty string 1`] = `
"<dim>expect(</><red>received</><dim>).toBeEmpty(</><dim>)</>

Expected value to be empty received:
  <red>\\"string\\"</>"
`;
